lab_3_ann_retriever/main.py

```diff
@@ -158,6 +199,16 @@ def build(self) -> bool:
             bool: True if built successfully, False in other case
         """
 
+        if not self._corpus or not isinstance(self._corpus, list):
+            return False
+        unique_terms = set(token for doc in self._corpus for token in doc)
+        self._vocabulary = sorted(unique_terms)
+        self._token2ind = {word: index for index, word in enumerate(self._vocabulary)}
+        self._idf_values = calculate_idf(self._vocabulary, self._corpus)
+        if self._idf_values is not None and self._vocabulary and self._token2ind:
```

Comment:
revert if statement; return False if something is wrong and return True at the end if everything is ok!

Reply:
Done